enclave_manager/Dockerfile

```diff
@@ -218,51 +218,26 @@ RUN mkdir -p build \
 
 WORKDIR /project/avalon/common/python
 
-RUN make
+RUN make && make install
 
 WORKDIR /project/avalon/common/crypto_utils
 
-RUN make
+RUN make && make install
 
 WORKDIR /project/avalon/sdk
 
-RUN make
+RUN make && make install
 
 WORKDIR /project/avalon/enclave_manager
 
-RUN echo "Building avalon enclave manager" \
-  && make
-
-# Build Final image and install dependent modules
-FROM base_image as final_image
-
-ENV TCF_HOME=/project/avalon
+RUN echo "Build and Install avalon enclave manager" \
+  && make && make install
 
-WORKDIR /project/avalon/
+FROM build_image as final_image
 
-COPY --from=build_image /project/avalon/config/tcs_config.toml /project/avalon/config/
-COPY --from=build_image /project/avalon/tc/sgx/trusted_worker_manager/enclave/build/lib/*.so /project/avalon/tc/sgx/trusted_worker_manager/enclave/build/lib/
-COPY --from=build_image /project/avalon/tc/sgx/trusted_worker_manager/enclave_untrusted/enclave_bridge/build/lib/*.so /project/avalon/tc/sgx/trusted_worker_manager/enclave_untrusted/enclave_bridge/build/lib/
-# Copy sgxsdk, sgxssl to final image.
-COPY --from=build_image /opt/intel/sgxsdk/bin/ /opt/intel/sgxsdk/bin
-COPY --from=build_image /opt/intel/sgxsdk/lib64/ /opt/intel/sgxsdk/lib64
-COPY --from=build_image /opt/intel/sgxssl/lib64/ /opt/intel/sgxssl/lib64
-
-# Copy Python build artifacts
-COPY --from=build_image /project/avalon/common/python/dist/*.whl dist/
-COPY --from=build_image /project/avalon/common/crypto_utils/dist/*.whl dist/
-COPY --from=build_image /project/avalon/sdk/dist/*.whl dist/
-COPY --from=build_image /project/avalon/enclave_manager/dist/*.whl dist/
-
-
-# Installing wheel file requires python3-pip package.
-# But python3-pip package will increase size of final docker image.
-# So remove python3-pip package and dependencies after installing wheel file.
-RUN apt-get update \
- && apt-get install -y -q python3-pip \
- && echo "Install Common Python, SDK and Enclave Manager packages\n" \
- && pip3 install dist/*.whl \
- && echo "Remove unused packages from image\n" \
+# python3-pip package will increase size of final docker image.
+# So remove python3-pip package and dependencies
+RUN echo "Remove unused packages from image\n" \
  && apt-get autoremove --purge -y -q python3-pip \
```

Comment:
Maybe there are more packages that are required during build only? (For example, please confirm if `ocamlbuild` is required, I see this also is installed in build_image)

Reply:
True, thats the reason we use to copy required packages from build stage to final stage earlier. But we have hit some blocker issue in HW mode which is due to optimization done in final stage.
Please note this is a temporary enable of SGX HW mode. We still need to fix the issue and bring back the optimization.
This can be tracked in git issue - https://github.com/hyperledger/avalon/issues/333

Reply:
Ok